### PR TITLE
Live: Simplify live interface

### PR DESCRIPTION
**Pull request by ryantxu**
The current live interface exposes a 'channel' interface -- this updates the main interface so that:

* the main interface returns observables
* query stream is managed internally to live server -- rather than a wrapper around it
* removes publish from the websocket interface -- we can do that over HTTP now

The interface is now:
```
export interface GrafanaLiveSrv {
  getConnectionState(): Observable<boolean>;
  getChannelInfo(address: LiveChannelAddress): Promise<LiveChannelConfig>;
  getStream<T>(address: LiveChannelAddress): Observable<LiveChannelEvent<T>>;
  getDataStream(options: LiveDataStreamOptions): Observable<DataQueryResponse>;
  getPresence(address: LiveChannelAddress): Promise<LiveChannelPresenceStatus>;
}
```

Previously it let you get a channnel, *then* subscribe:
```
getGrafanaLiveSrv().getChannel( addr ).getStream( ... )
```
This refactoring removes the intermediate `getChannel()` part.  This change will make it easier to delegate a shared cache to a shared worker if that proves to be a worthwhile performance/feature gain.

### CODE REVIEW

File: public/app/features/live/live.ts [hunk 1]
```diff
@@ -141,20 +161,163 @@ export class CentrifugeSrv implements GrafanaLiveSrv {
...
+  /**
+   * Watch for messages in a channel
+   */
+  getStream<T>(address: LiveChannelAddress): Observable<LiveChannelEvent<T>> {
+    return this.getChannel<T>(address).getStream();
+  }
+
+  /**
+   * Connect to a channel and return results as DataFrames
+   */
```
Comment (ryantxu) [inline public/app/features/live/live.ts:200]: NOTE:  this is just moved from an external function to one that is directly on the liveSrv

File: pkg/services/live/pushurl/values.go [hunk 2]
```diff
@@ -5,9 +5,10 @@ import (
 	"strings"
 )
 
+// StableSchemaFromValues assumes stable
 func StableSchemaFromValues(values url.Values) bool {
 	key := "gf_live_stable_schema"
-	return strings.ToLower(values.Get(key)) == "true" || values.Get(key) == "1"
+	return !(strings.ToLower(values.Get(key)) == "false" || values.Get(key) == "0")
```
Comment (ryantxu) [inline pkg/services/live/pushurl/values.go:11]: @FZambia -- this changes the default to assume things are stable and requires an explicit flag to say they are not.

Comment (FZambia) [reply pkg/services/live/pushurl/values.go:11]: Need to rename `gf_live_stable_schema` also then and fix caller's logic accordingly

File: public/app/features/live/live.ts [hunk 3]
```diff
@@ -141,20 +161,163 @@ export class CentrifugeSrv implements GrafanaLiveSrv {
...
+   *
+   * Join and leave messages will be sent to the open stream
+   */
+  getPresence(address: LiveChannelAddress): Promise<LiveChannelPresenceStatus> {
+    return this.getChannel(address).getPresence();
   }
 }
 
+// incremet the stream ids
+let streamCounter = 10;
```
Comment (FZambia) [inline public/app/features/live/live.ts:319]: Just curious - why 10 here?

Comment (ryantxu) [reply public/app/features/live/live.ts:319]: could be anything -- I'll change to zero

**Review by leeoniya — APPROVED** [review]
my approval amounts mostly to "does not appear to obviously break my shit"

**Review by FZambia — APPROVED** [review]

**Pull request merged**